skema/program_analysis/CAST/python/ts2cast.py

```diff
@@ -147,14 +155,92 @@ def visit_expression(self, node: Node):
 
         return expr_body
 
+    def visit_function_def(self, node: Node) -> FunctionDef:
+        ref = self.node_helper.get_source_ref(node)
+
+        name_node = get_first_child_by_type(node, "identifier")
+        name = self.visit(name_node)
+
+        # Create new variable context
+        self.variable_context.push_context()
+
+        parameters = get_children_by_types(node, "parameters")[0]
+        parameters = get_non_control_children(parameters)
+        
+        # The body of the function is stored in a 'block' type node
+        body = get_children_by_types(node, "block")[0].children
+        
+
+        func_params = []
+        for node in parameters:
```

Comment:
Be careful about overriding the symbol `node` in this manner since it is also the name of a function argument. 

This will change the value of `node` further down in the code if you try to access it again.

However, you can still use it in a list comprehension without changing the value of the top level symbol: 
```Pyhon
func_params = [node for node in parameters]  
```

Reply:
Oh yeah, I'll keep that in mind as that will most definitely be relevant later on. Coming up with variable names is hard..